infra/asttools.py

```diff
@@ -152,6 +158,23 @@ def double_visit_and_patches(self, tree):
         self.visit(tree)
         return (self.visit(tree), self.patches)
 
+def has_js_hide(decorator_list):
+    return any([
+        isinstance(dec, ast.Attribute) and dec.attr == "js_hide"
+        and isinstance(dec.value, ast.Name) and dec.value.id == "wbetools"
+        for dec in decorator_list
+    ])
+
+class ResolveJSHide(ast.NodeTransformer):
+    def visit_FunctionDef(self, cmd):
+        if any([
```

Comment:
This code duplicates `has_js_hide`. Did you mean to call it?

Reply:
Well, it would be circular. Maybe it would be good to call in the other direction but I thought that was maybe complicated. Can fix.

Reply:
Nothing seems to call `has_js__hide`...

Reply:
Oh, d'oh! Yes, I copied it over but then inlined it into `ResolveJSHide`. Removed the `has_js_hide` method..